### PR TITLE
feat: support `score()` in GeminiTextGenerator

### PR DESCRIPTION
**Pull request by ashleyxuu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes internal #343789535 🦕

### CODE REVIEW

File: bigframes/ml/llm.py [hunk 1]
```diff
@@ -732,6 +732,68 @@ def predict(
...
+
+        Returns:
+            bigframes.dataframe.DataFrame: The DataFrame as evaluation result.
+        """
+        if not self._bqml_model:
+            raise RuntimeError("A model must be fitted before score")
+
+        if self._bqml_model.model_name.startswith("gemini-1.5"):
+            raise NotImplementedError(
+                "Score is not supported for gemini-1.5 model. Please use gemini-pro-1.0 model instead."
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:777]: 1. Add doc to specify supported model types
2. As Tianxiang is working on the fix, plz add a TODO bug to track.
3. Just say 1.5 not supported is enough. No need to recommend 1.0.

File: tests/system/load/test_llm.py [hunk 2]
```diff
@@ -112,3 +112,44 @@ def test_llm_palm_score_params(llm_fine_tune_df_default_index):
         "evaluation_status",
     ]
     assert all(col in score_result_col for col in expected_col)
+
+
+@pytest.mark.flaky(retries=2)
```
Comment (GarrettWu) [inline tests/system/load/test_llm.py:117]: Unless really needed, add in small tests by default.

File: tests/system/load/test_llm.py [hunk 3]
```diff
@@ -112,3 +112,44 @@ def test_llm_palm_score_params(llm_fine_tune_df_default_index):
...
+    ).to_pandas()
+    score_result_col = score_result.columns.to_list()
+    expected_col = [
+        "bleu4_score",
+        "rouge-l_precision",
+        "rouge-l_recall",
+        "rouge-l_f1_score",
+        "evaluation_status",
+    ]
+    assert all(col in score_result_col for col in expected_col)
```
Comment (GarrettWu) [inline tests/system/load/test_llm.py:134]: Let's use https://github.com/googleapis/python-bigquery-dataframes/blob/main/tests/system/utils.py#L269 to check.

File: tests/system/load/test_llm.py (hunk 2 above)
Comment (ashleyxuu) [reply tests/system/load/test_llm.py]: Done.

File: tests/system/load/test_llm.py (hunk 3 above)
Comment (ashleyxuu) [reply tests/system/load/test_llm.py]: Done.

File: bigframes/ml/llm.py (hunk 1 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py]: Done.

**Review by GarrettWu — APPROVED** [review]

**Pull request merged**